site/src/benchmark_metadata/mod.rs

```diff
@@ -0,0 +1,74 @@
+use hashbrown::HashMap;
+use rust_embed::RustEmbed;
+
+use collector::compile::benchmark::category::Category;
+use collector::compile::benchmark::BenchmarkConfig;
+
+use crate::benchmark_metadata::metadata::SERIALIZED_SUITE_NAME;
+
+mod metadata;
+pub use metadata::ProfileMetadata;
+
+#[derive(Debug, Clone)]
+pub struct CompileBenchmarkMetadata {
+    pub perf_config: BenchmarkConfig,
+    pub release_metadata: ProfileMetadata,
+}
+
+/// The metadata of compile-time benchmarks is embedded directly within the binary using
+/// the `EmbeddedCompileBenchmarks` struct.
+#[derive(RustEmbed)]
+#[folder = "$OUT_DIR"]
+struct EmbeddedCompileBenchmarks;
+
+pub fn get_compile_benchmarks_metadata() -> &'static HashMap<String, CompileBenchmarkMetadata> {
+    lazy_static::lazy_static! {
+        static ref METADATA: HashMap<String, CompileBenchmarkMetadata> = load_compile_benchmark_metadata();
+    }
+    &METADATA
+}
+
+pub fn get_stable_benchmark_names() -> Vec<String> {
+    get_compile_benchmarks_metadata()
+        .iter()
+        .filter(|(_, metadata)| metadata.perf_config.category() == Category::Stable)
+        .map(|(key, _)| key.clone())
+        .collect()
+}
+
+fn load_compile_benchmark_metadata() -> HashMap<String, CompileBenchmarkMetadata> {
+    let serialized = EmbeddedCompileBenchmarks::get(SERIALIZED_SUITE_NAME)
+        .unwrap()
+        .data
+        .to_vec();
+    let metadata: metadata::CompileBenchmarkSuite = serde_json::from_slice(&serialized)
+        .expect("Cannot deserialize compile benchmarks metadata");
+    metadata
+        .benchmarks
+        .into_iter()
+        .map(|(name, metadata)| {
+            let metadata::CompileBenchmarkMetadata {
+                perf_config,
+                release_metadata,
+            } = metadata;
+            let perf_config: BenchmarkConfig =
+                serde_json::from_value(perf_config).expect("Cannot deserialize perf-config.json");
```

Comment:
I'm a bit worried that if we leave it optional, it will just silently be set to a wrong value. But you raise a good point. The error message should be improved, and I should also modify `collector download` to automatically generate the artifact type.

Reply:
If we only use it in the tooltip, it's probably fine. The filtering may be more impactful, and we could show missing data present in both lib/bin categories (with the help text explaining how to fix it). Though I'm not sure if using the filter would be very common, as we have basically 2 binaries and a bunch of simili-hello world.